### PR TITLE
Update Controllers to version 33.0.0

### PR DESCRIPTION
**Pull request by tommasini**
**Description**
Updating the controllers to version 33.0.0, will update the way Network Controller works
It was updated the PhishingControler
The CollectiblesController was renamed to NftController

**Test Cases**
Case1:
* Tested swaps on mainnet (uniswap) and custom network (Binance Smart Chain) (pancakeswap)
Case2:
* Tested transfer of collectible to myself on Mainnet
Case3:
* Tested token detection and nft Detection
Case4:
* Tested transfers between accounts, and the gas fee seems to be working and well defined (confirmed with etherscan and polyscan)
Case 5:
* Tested deep link with network changing
Case 6:
* Tested the activity view in different networks
Case 7:
* Tested switching accounts and verifying the balance on them (Also switching networks)
Case 8:
* Tested transfer collectables to other account on goerly
Case 9:
* Tested to create an new account it should change to it 

Would be good to do an overall test of the app.

**Issue**

Progresses #5092 

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/core/BackgroundBridge/BackgroundBridge.js [hunk 1]
```diff
@@ -89,7 +89,10 @@ export class BackgroundBridge extends EventEmitter {
       ),
     );
 
-    Engine.context.NetworkController.subscribe(this.sendStateUpdate);
+    Engine.controllerMessenger.subscribe(
+      `NetworkController:stateChange`,
```
Comment (sethkfman) [inline app/core/BackgroundBridge/BackgroundBridge.js:93]: Maybe we can create a const and reference throughout the codebase. I see this in a few locations

Comment (tommasini) [reply app/core/BackgroundBridge/BackgroundBridge.js:93]: Completely agree 💯

Comment (sethkfman) [inline app/core/BackgroundBridge/BackgroundBridge.js:93]: Missed one

File: app/core/BackgroundBridge/BackgroundBridge.js [hunk 2]
```diff
@@ -266,7 +269,10 @@ export class BackgroundBridge extends EventEmitter {
 
   onDisconnect = () => {
     this.disconnected = true;
-    Engine.context.NetworkController.unsubscribe(this.sendStateUpdate);
+    Engine.controllerMessenger.unsubscribe(
+      `NetworkController:stateChange`,
```
Comment (sethkfman) [inline app/core/BackgroundBridge/BackgroundBridge.js:273]: Missed one

File: app/core/EngineService.ts [hunk 3]
```diff
@@ -65,8 +65,15 @@ class EngineService {
...
         store.dispatch({ type: UPDATE_BG_STATE_KEY, key: name });
-      if (!key) Engine.context[name].subscribe(update_bg_state_cb);
-      else Engine.controllerMessenger.subscribe(key, update_bg_state_cb);
+      if (name !== 'NetworkController')
+        !key
+          ? Engine.context[name].subscribe(update_bg_state_cb)
+          : Engine.controllerMessenger.subscribe(key, update_bg_state_cb);
+      else
+        Engine.controllerMessenger.subscribe(
+          'NetworkController:stateChange',
```
Comment (sethkfman) [inline app/core/EngineService.ts:74]: Missed one

**Comment by Gudahtt** [comment]
I have created a PR to account for the PhishingController breaking changes included in the v31 and v32 controllers releases: #5215

File: app/core/Engine.js [hunk 4]
```diff
@@ -84,61 +85,77 @@ class Engine {
...
-          getAccounts: (
-            end: (arg0: null, arg1: any[]) => void,
-            payload: { hostname: string | number },
+        state: initialState.networkController,
+        messenger: this.controllerMessenger.getRestricted({
+          name: 'NetworkController',
+          allowedEvents: [
+            AppConstants.NETWORK_STATE_CHANGE_EVENT,
+            'NetworkController:providerChange',
+          ],
```
Comment (Gudahtt) [inline app/core/Engine.js:97]: Nit: I don't think any allowed events are needed here. It should automatically have permission to use any events or actions under its own namespace

```suggestion
          allowedEvents: [],
```

File: app/core/Engine.js [hunk 5]
```diff
@@ -378,14 +416,14 @@ class Engine {
       const {
         CollectiblesController: collectibles,
         KeyringController: keyring,
-        NetworkController: network,
         TransactionController: transaction,
       } = this.context;
 
       collectibles.setApiKey(process.env.MM_OPENSEA_KEY);
-      network.refreshNetwork();
```
Comment (Gudahtt) [inline app/core/Engine.js:386]: Interesting - even in the older version of the controllers package, this was a private method. It seems that setting `providerConfig` has basically the same effect as calling this, so I agree that it is not needed.

**Review by Gudahtt — COMMENTED** [review]
The NetworkController changes all look good to me!

File: app/core/Engine.js (hunk 4 above)
Comment (tommasini) [reply app/core/Engine.js:97]: Confirmed

**Comment by sethkfman** [comment]
@SocketSecurity ignore @ethereumjs/rlp@4.0.0 rlp@2.2.6 rlp@3.0.0

**Review by sethkfman — APPROVED** [review]
LGTM

**Comment by Gudahtt** [comment]
This PR still needs to be merged into this: https://github.com/MetaMask/metamask-mobile/pull/5215

Otherwise the phishing configuration will never be updated

File: app/store/migrations.js [hunk 6]
```diff
@@ -300,6 +300,32 @@ export const migrations = {
...
+    state.engine.backgroundState.PreferencesController.useNftDetection =
+      state.engine.backgroundState.PreferencesController.useCollectibleDetection;
+    delete state.engine.backgroundState.PreferencesController
+      .useCollectibleDetection;
+
+    return state;
+  },
 };
 
 export const version = 11;
```
Comment (Gudahtt) [inline app/store/migrations.js:331]: Oops, I think I missed this!
```suggestion
export const version = 12;
```

Comment (tommasini) [reply app/store/migrations.js:331]: Oops, escaped me as well! Thanks for backing here!

**Comment by Gudahtt** [comment]
I'm seeing this warning during the `setup` step in CI:

```
Warning: patch-package detected a patch file version mismatch

  Don't worry! This is probably fine. The patch was still applied
  successfully. Here's the deets:

  Patch file created for

    @metamask/controllers@30.3.0

  applied to

    @metamask/controllers@33.0.0
  
  At path
  
    node_modules/@metamask/controllers

  This warning is just to give you a heads-up. There is a small chance of
  breakage even though the patch was applied successfully. Make sure the package
  still behaves like you expect (you wrote tests, right?) and then run

    patch-package @metamask/controllers

  to update the version in the patch file name and make this warning go away.

```

Seems like that patch file needs an update

**Review by Gudahtt — APPROVED** [review]

**Comment by chrisleewilcox** [comment]
Swaps:
Polygon
  MATIC => WMATIC => Edit gas fee to Market: PASS
  MATIC => UNI: PASS
  USDT => LINK: PASS
  UNI => DAI: PASS
  Cancel transaction: having problems cancelling.  Getting console warning on cancel: https://recordit.co/9xDj8XntwW

**Comment by chrisleewilcox** [comment]
NFTs

Autodetect NFTs:
- ETH Mainnet: PASS

**Comment by chrisleewilcox** [comment]
Wallet view:

Token fiat balance for asset is incorrect.  See [here](image).

**Review by chrisleewilcox — CHANGES_REQUESTED** [review]
Please review comments.

**Comment by chrisleewilcox** [comment]
Send Tokens:

ETH QRC:
ETH Deeplink:
Polygon QRC:
Polygon Deeplink:
BNB QRC:
BNB Deeplink:
AVAX QRC:
AVAX Deeplink:
Arbitrum QRC:
Arbitrum Deeplink:
Optimism QRC:
Optimism Deeplink:

**Comment by Gudahtt** [comment]
Are these errors reproducible on `main`? Also, when these errors were encountered, did you recently switch networks?

It's hard to see how these changes would cause cancellation failures, edit limit failures, balance failures, etc. unless maybe there's a bug related to network switching. The code changes barely touch the transaction controller or asset management parts at all.

**Comment by chrisleewilcox** [comment]
dApps

Uniswap:

- swaps:

Sushiswap swap:

- swaps:

pancake swap:

- swaps:

1inch:

- swaps:

**Review by chrisleewilcox — APPROVED** [review]
LGTM

**Pull request merged**